src/Workspaces/VisualBasic/Portable/CodeCleanup/VisualBasicCodeCleanerService.vb

```diff
@@ -3,23 +3,90 @@
         Inherits AbstractCodeCleanerService
 
         Private Shared ReadOnly s_defaultProviders As ImmutableArray(Of ICodeCleanupProvider) = ImmutableArray.Create(Of ICodeCleanupProvider)(
-                New AddMissingTokensCodeCleanupProvider(),
-                New FixIncorrectTokensCodeCleanupProvider(),
-                New ReduceTokensCodeCleanupProvider(),
-                New NormalizeModifiersOrOperatorsCodeCleanupProvider(),
-                New RemoveUnnecessaryLineContinuationCodeCleanupProvider(),
-                New CaseCorrectionCodeCleanupProvider(),
-                New SimplificationCodeCleanupProvider(),
-                New FormatCodeCleanupProvider())
+            New AddMissingTokensCodeCleanupProvider(),
+            New FixIncorrectTokensCodeCleanupProvider(),
+            New ReduceTokensCodeCleanupProvider(),
+            New NormalizeModifiersOrOperatorsCodeCleanupProvider(),
+            New RemoveUnnecessaryLineContinuationCodeCleanupProvider(),
+            New CaseCorrectionCodeCleanupProvider(),
+            New SimplificationCodeCleanupProvider(),
+            New FormatCodeCleanupProvider())
 
         Public Overrides Function GetDefaultProviders() As ImmutableArray(Of ICodeCleanupProvider)
             Return s_defaultProviders
         End Function
+
+        Protected Overrides Function GetSpansToAvoid(root As SyntaxNode) As ImmutableArray(Of TextSpan)
+            ' We don't want to touch nodes in the document that have syntax errors on them and which
+            ' contain multi-line string literals.  It's quite possible that there is some string 
+            ' literal on a previous line that was intended to be terminated on that line, but which
+            ' wasn't.  The string may then have terminated on this line (because of the start of 
+            ' another literal) causing the literal contents to then be considered code.  We don't 
+            ' want to cleanup 'code' that the user intends to be the content of a string literal.
+
+            Dim result = ArrayBuilder(Of TextSpan).GetInstance()
+
+            ProcessNode(root, result)
+
+            Return result.ToImmutableAndFree()
+        End Function
+
+        Private Sub ProcessNode(node As SyntaxNode, result As ArrayBuilder(Of TextSpan))
+            If Not node.ContainsDiagnostics Then
+                ' Don't bother going down nodes that don't have any syntax errors in them.
+                Return
+            End If
+
+            For Each child In node.ChildNodesAndTokens()
+                If child.IsNode Then
+                    ProcessNode(child.AsNode(), result)
+                Else
+                    ProcessToken(child.AsToken(), result)
+                End If
+            Next
+        End Sub
+
+        Private Sub ProcessToken(token As SyntaxToken, result As ArrayBuilder(Of TextSpan))
+            If token.ContainsDiagnostics Then
+                Dim parentMultiLineNode = GetMultiLineContainer(token.Parent)
```

Comment:
For each token, we walk up and then back down to all the Descendant tokens?  Is there something better we can do here?